modules/artifact/src/main/java/org/mule/module/artifact/classloader/JdbcResourceReleaser.java

```diff
@@ -264,4 +276,71 @@ private void shutdownMySqlConnectionCleanupThreads(Class<?> classAbandonedConnec
     throw new ClassNotFoundException("No MySql's AbandonedConnectionCleanupThread class was found");
   }
 
+  private void disposeOracleDriverThreads() {
+    try {
+      if (JDBC_RESOURCE_RELEASER_AVOID_DISPOSE_ORACLE_THREADS) {
+        return;
+      }
+
+      Thread[] threads = new Thread[activeCount()];
+      try {
+        enumerate(threads);
+      } catch (Throwable t) {
+        logger
+            .debug("An error occurred trying to obtain available Threads. Thread cleanup will be skipped.", t);
+        return;
+      }
+
+      /* IMPORTANT: this is done to avoid metaspace OOM caused by oracle driver
+      thread leak. This is only meant to stop TimerThread threads spawned
+      by oracle driver's HAManger class. This timer cannot be fetched
+      by reflection because, in order to do so, other oracle dependencies
+      would be required.
+      * */
+      for (Thread thread : threads) {
+        if (isThreadApplicationTimerThread(thread)) {
+          try {
+            thread.stop();
+            thread.interrupt();
+          } catch (Throwable e) {
+            logger
+                .debug("An error occurred trying to close the '" + thread.getName() + "' Thread. This might cause memory leaks.",
+                       e);
+          }
+        }
+      }
+    } catch (Exception e) {
+      logger.debug(e.getMessage());
```

Comment:
this should log not only the message, but the context in which it happened and the entire exception